acme/db/nosql/account.go

```diff
@@ -259,21 +259,42 @@ func (db *DB) DeleteExternalAccountKey(ctx context.Context, provisionerName, key
 }
 
 // GetExternalAccountKeys retrieves all External Account Binding keys for a provisioner
-func (db *DB) GetExternalAccountKeys(ctx context.Context, provisionerName string) ([]*acme.ExternalAccountKey, error) {
+func (db *DB) GetExternalAccountKeys(ctx context.Context, provisionerName, cursor string, limit int) ([]*acme.ExternalAccountKey, string, error) {
 	entries, err := db.db.List(externalAccountKeyTable)
 	if err != nil {
-		return nil, err
+		return nil, "", err
+	}
+
+	// set sane limits; based on the Admin API limits
+	switch {
+	case limit <= 0:
+		limit = 20
+	case limit > 100:
+		limit = 100
 	}
 
+	foundCursorKey := false
 	keys := []*acme.ExternalAccountKey{}
-	for _, entry := range entries {
+	for _, entry := range entries { // entries is sorted alphabetically on the key (ID) of the EAK; no need to sort this again.
 		dbeak := new(dbExternalAccountKey)
 		if err = json.Unmarshal(entry.Value, dbeak); err != nil {
-			return nil, errors.Wrapf(err, "error unmarshaling external account key %s into ExternalAccountKey", string(entry.Key))
+			return nil, "", errors.Wrapf(err, "error unmarshaling external account key %s into ExternalAccountKey", string(entry.Key))
 		}
 		if dbeak.Provisioner != provisionerName {
 			continue
 		}
+		// skip the IDs not matching the cursor to look for in the sorted list.
+		if cursor != "" && !foundCursorKey && cursor != dbeak.ID {
```

Comment:
This logic was kinda difficult for me to follow and I wonder if it wouldn't be better to spit up the conditionals a bit:

```
if cursor != "" && !foundCursorKey {
    if cursor == dbeak.ID {
        founcCursorKey = true
    } else {
        continue
    }
}
```

This also cuts down on a few comparisons in a loop.

Reply:
Agree, your version is simpler. Fixed in 11a7f011778a4f771f838cef4e8635a2e73cea25.